ouroboros-consensus/src/Ouroboros/Consensus/Mempool/ImplPure.hs

```diff
@@ -0,0 +1,654 @@
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE LambdaCase           #-}
+{-# LANGUAGE NamedFieldPuns       #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE StandaloneDeriving   #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Ouroboros.Consensus.Mempool.ImplPure (
+    MempoolArgs(..)
+  , InternalState(..)
```

Comment:
Can we make `InternalState` abstract so that nobody can fiddle with the internals? If necessary.

When you do that, be sure to add the following comment:
```suggestion
  , InternalState -- opaque
```
That's a convention we have that makes it clear that it is intentionally opaque, not because we forgot it.